Sources/HipMobileUI/ViewModels/Pages/UserRatingPageViewModel.cs

```diff
@@ -93,59 +92,52 @@ public UserRatingPageViewModel(Exhibit exhibit)
 
         private async void SetUserRatingUi()
         {
-            UserRating userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRatingAsync(exhibit.IdForRestApi);
+            var userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRatingAsync(exhibit.IdForRestApi);
             SetAverageAndCountRating(userRating.Average, userRating.Count);
             SetStarImages(userRating.Average);
             SetRatingBars(userRating.RatingTable, userRating.Count);
-            RatingStar1 = ImgStarEmpty;
-            RatingStar2 = ImgStarEmpty;
-            RatingStar3 = ImgStarEmpty;
-            RatingStar4 = ImgStarEmpty;
-            RatingStar5 = ImgStarEmpty;
+            SetPreviousUserRating(userRating.Count);
+        }
+
+        private async void SetPreviousUserRating(int count)
+        {
+            var prevRating = 0;
+            if (Settings.IsLoggedIn && count > 0)
+            {
+                prevRating = await IoCManager.Resolve<IUserRatingManager>().GetPreviousUserRatingAsync(exhibit.IdForRestApi);
+                SetRatingStars(5);
```

Comment:
What is this line good for?

Reply:
The if-statement tests, if the user is logged in and if the number of user ratings is bigger than 0 (the user can't has given a rating when the rating count is 0).
 The SetRatingStars(5) is needed, because of the way the stars are set. I have added some documentation for that method.

Reply:
Even with the added documentation, I don't understand how it works. In my opinion, the method should always behave the same, independent of the previous rating - that is, simply color the appropriate starts yellow and the others white. Is this not possible or do I misunderstand something?

Reply:
This would be the easiest way to implement this method, but that doesn't look good. The images are flickering each time they are changed. Because of that I have minimezed the amout of changes needed to show the correct stars.

Reply:
You could avoid flickering by not calling a rating star's setter when it's not needed to avoid flickering:

```C#
var ratingStar1Img = <Image for the first rating star>;
if (RatingStar1 != img) RatingStar1 = img;
var ratingStar2Img = <Image for the second rating star>;
if (RatingStar1 2= img) RatingStar2 = img;
[...]
```

So in `SetRatingStars`, you could find out the appropriate image for each star, but only call its setter if needed. Would that work?